ioos_qc/qartod.py

```diff
@@ -498,19 +499,26 @@ def run_test(test_threshold, flag_value):
 
     return flag_arr.reshape(original_shape)
 
+
 def attenuated_signal_test(inp : Sequence[N],
-                           threshold : Tuple[N, N],
-                           check_type : str = 'std'
+                           tinp : Sequence[N],
+                           suspect_threshold: N,
+                           fail_threshold: N,
+                           test_period: N = None, 
+                           check_type : str = 'std',
+                           *args,
+                           **kwargs,
                            ) -> np.ma.MaskedArray:
     """Check for near-flat-line conditions using a range or standard deviation.
 
     Missing and masked data is flagged as UNKNOWN.
 
     Args:
         inp: Input data as a numeric numpy array or a list of numbers.
-        threshold: 2-tuple representing the minimum thresholds to use for SUSPECT
-            and FAIL checks. The smaller of the two values is used in the SUSPECT
-            tests and the greater of the two values is used in the FAIL tests.
+        tinp: Time data as a numpy array of dtype `datetime64`.
```

Comment:
This should handle either datetimes or epoch seconds represented as integers. This will need some unit tests too -- see `QartodFlatLineTest` for an example